sechub-integrationtest/src/main/java/com/mercedesbenz/sechub/integrationtest/internal/SecHubClientExecutor.java

```diff
@@ -342,13 +343,30 @@ private File resolveExistingExecutableAndAppendAdditionalCommands() {
 
         if (sechubClientBinaryPath == null) {
             String parentFolder = "sechub-cli/build/go/platform/"; // when not set we use build location
-            String sechubExeName = null;
-            if (TestUtil.isWindows()) {
-                sechubExeName = "sechub.exe";
-                parentFolder += "windows-386";
+            String sechubExeName = "sechub";
+
+            String osArch = SystemUtils.OS_ARCH;
+            boolean is64 = osArch.contains("64");
+            boolean isArm = osArch.contains("arm") || osArch.contains("aarch");
```

Comment:
We can leave a comment in the PR to upstream repo about `ArchUtils` (or code comment here), so that SecHub maintainers know this logic can soon be replaced with nice call to a library.

Reply:
In theory, once a new version of Apache Commons Lang with [this commit](https://github.com/apache/commons-lang/commit/16d587d7) is released, we could do something like this:
```java
Processor processor = ArchUtils.getProcessor();
boolean is64 = processor.is64Bit();
boolean isArm = processor.isAarch64();
```
and everything should work on M1 Macbooks. However, there is a problem in this line: `boolean isArm = processor.isAarch64()`. For now `ArchUtils` only knows how to handle the string `aarch64` and SecHub also has binaries for `linux-arm64` and `linux-arm`. I'm not sure what string would be returned by `System.getProperty("os.arch")` in those cases, but I feel like there's more options than just `aarch64`?